server/src/main/java/io/druid/client/CachingClusteredClient.java

```diff
@@ -169,34 +186,64 @@ public CachingClusteredClient(
     return new SpecificQueryRunnable<>(queryPlus, responseContext).run(timelineConverter);
   }
 
+  private <T> Sequence<T> runAndMergeWithTimelineChange(
+      final Query<T> query,
```

Comment:
Passing both query and queryPlus contributes to ambiguity. query is queryPlus.getQuery().

Reply:
good call, fixing